### ISSUE TITLE
get_funded_wallet doesn't set the script_pubkey for its funding address in the db

### ISSUE THREAD

**Issue by davemo88** [bitcoindevkit/bdk#303]
`get_funded_wallet` calls `populate_test_db!` like this:
```
        let txid = crate::populate_test_db!(
            wallet.database.borrow_mut(),
            testutils! {
                @tx ( (@external descriptors, 0) => 50_000 ) (@confirmations 1)
            },
            Some(100)
        );
```
but `populate_test_db!` doesn't know how the address created in `@tx` was derived, so it can't set the corresponding script pubkey in the db.

I patched it up [like this](https://github.com/bitcoindevkit/bdk/compare/master...davemo88:get_funded_wallet) but it might be better to change the `populate_test_db!` macro signature and do it there.

**Comment by LLFourn** [comment]
@davemo88 showed that just calling `get_new_address` solves this problem:

```
  #[test]
    fn test_get_psbt_input() {
        // this should grab a known good utxo and set the input
        let (wallet, _, _) = get_funded_wallet(get_test_wpkh());
        wallet.get_new_address().unwrap();
        for utxo in wallet.list_unspent().unwrap() {
            let psbt_input = wallet.get_psbt_input(utxo, None, false).unwrap();
            assert!(psbt_input.witness_utxo.is_some() || psbt_input.non_witness_utxo.is_some());
        }
    }
```

I am a bit out of my depth here so we need @afilini to chime in the best way to address it. It feels like there should be some layer that guarantees that if there is a UTXO in the db then the script pubkey is too but it looks like we don't have that atm and populate_test_db just doesn't do the job of properly populating the test db.

**Comment by notmandatory** [comment]
Fixed in #310.
